src/query/sql/src/planner/binder/join.rs

```diff
@@ -43,6 +43,13 @@ use crate::plans::Scalar;
 use crate::plans::ScalarExpr;
 use crate::BindContext;
 
+pub struct JoinConditions {
+    pub(crate) left_conditions: Vec<Scalar>,
+    pub(crate) right_conditions: Vec<Scalar>,
+    pub(crate) non_equi_conditions: Vec<Scalar>,
+    pub(crate) other_conditions: Vec<Scalar>,
```

Comment:
What's the difference between `other_conditions` and `non_equi_conditions`?

Reply:
`other_conditions` can be pushed down. 
such as `select * from t left join t1 on t1.a > 1`, `t1.a > 1` is a `other_condition` and it can be pushed down.

`other_conditions` only exist in logical join, they'll be eliminated in the optimizer.

`non_equi_conditions` are neither `equi_conditions` nor conditions that can be pushed down. They should keep in `PhysicalJoin` and execute in `HashJoin` or we'll get the wrong results.